lib/services/template-service.ts

```diff
@@ -1,167 +1,85 @@
-import * as path from 'path';
-import * as fs from 'fs';
-import * as childProcess from 'child_process';
+import * as childProcess from "child_process";
+import * as fs from "fs";
+import * as path from "path";
 
-const request = require('request-promise');
-const Config = require('../../config.js');
-const Backup = require('../../consts/backup-data');
-const NodeCache = require("node-cache");
-const tmpCache = new NodeCache();
-const _indexof = require('lodash.indexof');
-const _sortby = require('lodash.sortby');
-
-export class TemplateService implements ITemplateService {
-    constructor() {
-    }
+import { Config } from "../shared/config";
+import util from "../shared/util";
 
-    public _sortTmpData(templates: Array<any>) {
-        let flavOrder: Array<string> = ['JavaScript', 'TypeScript', 'Angular & TypeScript'],
-            typeOrder: Array<string> = ['Blank', 'Navigation Drawer', 'Tabs', 'Master-Detail with Firebase', 'Master-Detail with Kinvey'],
-            sortedByType: Array<any>,
-            sortByFlav: Array<any>;
-
-        sortedByType = _sortby(templates, function (temp: any) {
-            return _indexof(typeOrder, temp.displayName);
-        });
-
-        sortByFlav = _sortby(sortedByType, function (temp: any) {
-            return _indexof(flavOrder, temp.templateFlavor);
-        });
+const request = require("request-promise");
+// tslint:disable-next-line:variable-name
+const NodeCache = require("node-cache");
 
-        return sortByFlav;
-    }
+// tslint:disable-next-line:variable-name
+const Backup = require("../../consts/backup-data");
 
-    public _getTemplatesNames() {
-        let repos: Array<any> = [
-            'template-drawer-navigation',
-            'template-tab-navigation',
-            'template-master-detail',
-            'template-blank',
-            'template-drawer-navigation-ts',
-            'template-master-detail-ts',
-            'template-blank-ts',
-            'template-tab-navigation-ts',
-            'template-drawer-navigation-ng',
-            'template-tab-navigation-ng',
-            'template-master-detail-ng',
-            'template-blank-ng',
-            'template-master-detail-kinvey',
-            'template-master-detail-kinvey-ng',
-            'template-master-detail-kinvey-ts'
-        ];
+const tmpCache = new NodeCache();
 
-        return new Promise(function (resolve, reject) {
-            if (!repos || typeof repos === 'undefined') {
-                reject('Error retrieving Template Name');
-            } else {
-                resolve(repos);
-            }
-        });
-    }
+export class TemplateService implements ITemplateService {
+    tmpPackageJsonFromSrc(templateName: string) {
```

Comment:
"I would go much further with the refactoring, but I would like to talk with Bobi first"